citest/json_predicate/binary_predicate.py

```diff
@@ -146,6 +150,83 @@ def __call__(self, context, value):
                            path_value=PathValue('', value), valid=valid)
 
 
+class DictMatchesPredicate(BinaryPredicate):
+  """Implements binary predicate comparison predicates against dict values.
+
+  The dictionary value is in the form {field : pred} where each element
```

Comment:
I'd like to, but not in this PR. For now you'd have to use path predicates for sparse trees. I think the cases I've seen so far were only one or two levels deep, so thought I could get away without it for now.